Uplift/Views/WeeklyWorkoutTrackerView.swift

```diff
@@ -0,0 +1,152 @@
+                    }
+
+                    // Workout day circles
+                    HStack(spacing: spacing) {
+                        ForEach(0..<7, id: \.self) { index in
+                            ZStack {
+                                Circle()
+                                    .fill(Color(.systemGray6))
+                                    .frame(width: circleSize, height: circleSize)
+
+                                if workoutDays[index] {
+                                    Circle()
+                                        .fill(Constants.Colors.yellow)
+                                        .frame(width: circleSize, height: circleSize)
+                                        .opacity(animationProgress > Double(index) ? 1 : 0)
+                                }
+
+                                if workoutDays[index] {
+                                    Image(systemName: "checkmark")
+                                        .font(.system(size: 14, weight: .bold))
+                                        .foregroundColor(.black)
+                                        .opacity(animationProgress > Double(index) ? 1 : 0)
+                                }
+                            }
+                        }
+                    }
+                }
+
+                HStack(spacing: spacing) {
+                    ForEach(0..<7, id: \.self) { index in
+                        Text("\(25 + index)")
+                            .font(Constants.Fonts.labelNormal)
+                            .frame(width: circleSize, height: 20)
+                            .foregroundColor(Constants.Colors.black)
+                    }
+                }
+            }
+        }
+        .padding(.top, 5)
+        .padding(.bottom, 15)
+        .onAppear {
+            if viewModel.workoutHistory.isEmpty {
+                viewModel.fetchUserProfile()
+            }
+        }
+        .onReceive(viewModel.$workoutHistory) { workouts in
+            if !workouts.isEmpty {
+                determineWorkoutDays()
+                animateWorkouts()
+            }
+        }
+    }
+
+    /// Determines which days of the week have completed workouts and updates the UI accordingly
+    private func determineWorkoutDays() {
+        let workoutDaysSet = Set(viewModel.workoutHistory.compactMap { workout -> Int? in
+            guard let date = Date.fromString(workout.date) else { return nil }
+            return Calendar.current.component(.day, from: date)
+        })
+
+        for i in 0..<7 {
+            let day = 25 + i
+            workoutDays[i] = workoutDaysSet.contains(day)
+        }
+    }
+
+    /// Animates the workout day indicators sequentially from left to right.
+    private func animateWorkouts() {
+        animationProgress = 0
+
+        for i in 0..<7 {
+            DispatchQueue.main.asyncAfter(deadline: .now() + Double(i) * (animationDuration + delayBetweenDays)) {
```

Comment:
DispatchQueue is totally valid to use but it may be a good idea to use a Task here instead since it is the newer Swift Concurrency Framework. So this would looking something like:
```

Task { @MainActor in
    try await Task.sleep(for: .seconds(<amount of time here>))
    // The task you want to execute after specified amount of time goes here
}
```